OMCompiler/Compiler/SimCode/SimCodeUtil.mo

```diff
@@ -5036,7 +5041,10 @@ algorithm
         v1 := BackendVariable.setVarKind(v1, BackendDAE.JAC_VAR());
         simVar := dlowvarToSimvar(v1, NONE(), inAllVars);
         simVar.index := inResIndex;
-        resIndex := inResIndex + 1;
+        resIndex := match Flags.isSet(Flags.NF_SCALARIZE)
```

Comment:
Wouldn't a simple if-else statement be more clear? But that's only my personal style. Feel free to ignore this.
```
if Flag.isSet(Flag.NF_SCALARIZE) then
  resIndex := inResIndex + 1;
else
  resIndex := inResIndex + Types.getDimensionProduct(simVar.type_);
end if;
```

Reply:
Or even just:
```
resIndex := inResIndex + if Flag.isSet(Flag.NF_SCALARIZE) then 1 else Types.getDimensionProduct(simVar.type_);
```